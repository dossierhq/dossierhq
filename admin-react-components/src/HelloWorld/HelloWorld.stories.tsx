import React from 'react';
import { HelloWorld } from './HelloWorld';

export default {
  title: 'Components/HelloWorld',
  component: HelloWorld,
};

export function Normal(): JSX.Element {
  return <HelloWorld />;
}
