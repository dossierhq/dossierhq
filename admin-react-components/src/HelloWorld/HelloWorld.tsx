import React from 'react';

export function HelloWorld(): JSX.Element {
  return <div>Hello world</div>;
}
