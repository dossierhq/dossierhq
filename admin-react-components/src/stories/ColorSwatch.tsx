import React, { useCallback, useState } from 'react';

interface ColorSwatchProps {
  className: string;
  disabled?: boolean;
}

export function ColorSwatch({ className }: ColorSwatchProps) {
  const [colors, setColors] = useState<null | { backgroundColor: string; color: string }>(null);

  const onRefChanged = useCallback(
    (ref) => {
      if (ref) {
        const { backgroundColor, color } = getComputedStyle(ref);
        setColors({ backgroundColor, color });
      }
    },
    [setColors]
  );

  return (
    <div
      className={`dd ${className} has-background text-body1`}
      ref={onRefChanged}
      style={{
        width: '150px',
        height: '150px',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
      }}
    >
      {className}
      <br />
      {colors ? (
        <>
          {colors.backgroundColor}
          <br />
          {colors.color}
        </>
      ) : null}
    </div>
  );
}

interface ColorSwatchGroupProps {
  className: string;
}

export function ColorSwatchGroup({ className }: ColorSwatchGroupProps) {
  return (
    <div style={{ display: 'flex', flexDirection: 'row' }}>
      <ColorSwatch className={className} />
      <ColorSwatch className={`${className} is-disabled`} />
      <ColorSwatch className={`${className} is-hover`} />
      <ColorSwatch className={`${className} is-active`} />
    </div>
  );
}
