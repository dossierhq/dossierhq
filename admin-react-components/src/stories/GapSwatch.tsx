import React from 'react';
import { gapClassName, joinClassNames } from '../utils/ClassNameUtils';

interface GapSwatchProps {
  size: SpacingSize;
}

export function GapSwatch({ size }: GapSwatchProps): JSX.Element {
  return (
    <div
      className={joinClassNames('dd flex-row', gapClassName(size))}
      style={{ border: 'solid 1px black' }}
    >
      {[...new Array(5)].map((_, index) => (
        <div
          key={index}
          className="dd text-caption p-1"
          style={{
            width: '30px',
            height: '30px',
            backgroundColor: 'turquoise',
          }}
        >
          {gapClassName(size)}
        </div>
      ))}
    </div>
  );
}
