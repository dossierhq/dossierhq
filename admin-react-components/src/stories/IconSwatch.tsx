import React from 'react';
import type { IconProps } from '../';
import { Icon, IconTypes } from '../';

export function IconSwatch({ icon }: IconProps): JSX.Element {
  return (
    <div className="dd has-background text-body1">
      {icon}
      <br />
      <div style={{ display: 'flex', flexDirection: 'row' }}>
        <span
          style={{
            outline: '1px dashed black',
            display: 'inline-block',
          }}
        >
          <Icon icon={icon} />
        </span>
        {['primary', 'danger'].map((bg) => (
          <div key={bg} className={`dd has-background bg-${bg}`}>
            <Icon icon={icon} />
          </div>
        ))}
      </div>
    </div>
  );
}

export function AllIconSwatches(): JSX.Element {
  return (
    <>
      {IconTypes.map((icon) => (
        <IconSwatch key={icon} icon={icon} />
      ))}
    </>
  );
}
