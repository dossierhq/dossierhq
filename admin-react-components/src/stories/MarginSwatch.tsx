import React from 'react';
import type { SpacingSize } from '..';
import { joinClassNames } from '../utils/ClassNameUtils';

interface MarginSwatchProps {
  className: string;
}

export function MarginSwatch({ className }: MarginSwatchProps): JSX.Element {
  return (
    <div style={{ border: 'solid 1px black' }}>
      <p
        className={joinClassNames('dd text-caption', className)}
        style={{
          width: '30px',
          height: '30px',
          backgroundColor: 'green',
          textAlign: 'center',
          padding: '0.5rem',
        }}
      >
        {className}
      </p>
    </div>
  );
}

interface MarginSwatchGroupProps {
  size: SpacingSize;
}

export function MarginSwatchGroup({ size }: MarginSwatchGroupProps): JSX.Element {
  return (
    <div className="dd flex-row g-2" style={{ alignItems: 'center' }}>
      <MarginSwatch className={`m-${size}`} />
      <MarginSwatch className={`mx-${size}`} />
      <MarginSwatch className={`my-${size}`} />
      <MarginSwatch className={`ml-${size}`} />
      <MarginSwatch className={`mt-${size}`} />
      <MarginSwatch className={`mr-${size}`} />
      <MarginSwatch className={`mb-${size}`} />
    </div>
  );
}
