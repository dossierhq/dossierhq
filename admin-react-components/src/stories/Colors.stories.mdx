import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { ColorSwatchGroup } from './ColorSwatch';

<Meta title="Colors" />

## Colors

<ColorSwatchGroup className="" />
<ColorSwatchGroup className="bg-blank1" />
<ColorSwatchGroup className="bg-blank2" />
<ColorSwatchGroup className="bg-primary" />
<ColorSwatchGroup className="bg-danger" />
