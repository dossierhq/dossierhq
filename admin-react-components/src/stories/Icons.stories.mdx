import { Meta } from '@storybook/addon-docs/blocks';
import { AllIconSwatches } from './IconSwatch';

<Meta title="Icons" />

## Icons

<AllIconSwatches />
