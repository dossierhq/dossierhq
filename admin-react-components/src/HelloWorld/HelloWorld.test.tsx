import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Normal } from './HelloWorld.stories';

test('Normal', () => {
  const tree = renderer.create(<Normal />).toJSON();
  expect(tree).toMatchSnapshot();
});
