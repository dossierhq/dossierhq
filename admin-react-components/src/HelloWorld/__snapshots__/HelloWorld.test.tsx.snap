// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normal 1`] = `
<div>
  Hello world
</div>
`;
